photon-lib/src/main/scala/com/linkedin/photon/ml/util/Timed.scala

```diff
@@ -70,8 +70,24 @@ object Timed {
     val t0 = System.nanoTime()
     val res = f
     val t1 = System.nanoTime()
-    val duration = FiniteDuration(t1 - t0, TimeUnit.NANOSECONDS)
-    logger.info(s"$msg: executed in ${duration.toUnit(units)} $units")
+    val duration = roundToSignificantFigures(FiniteDuration(t1 - t0, TimeUnit.NANOSECONDS).toUnit(units))
+    logger.info(s"$msg: executed in $duration $units")
     res
   }
+
+  /**
+   * A private method to reduce the "noise" and round to show only a limited number of significant figures.
+   *
+   * @param num The number to round
+   * @param n The number of significant figures to keep
+   * @return A Double with only n significant figures
+   */
+  def roundToSignificantFigures(num: Double, n: Int = 2): Double = {
+    if (num == 0) return 0
+    val d = Math.ceil(Math.log10(if (num < 0) -num else num))
+    val power = n - d.toInt
+    val magnitude = Math.pow(10, power)
+    val shifted = num * magnitude.round
+    shifted.toInt / magnitude
+  }
```

Comment:
I don't think so. In the case of 0.0031415926535, I want to see 0.0031, not 0.0, and the code you suggest returns 0.0. We can combine with scientific notation, so that we see 3.1e-3.